packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/implementations/method_channel_document_reference.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show visibleForTesting;
+
+import '../interfaces/document_reference_platform.dart';
+
+class MethodChannelDocumentReference extends DocumentReferencePlatform {
+  MethodChannelDocumentReference() {
+    channel.setMethodCallHandler(_callHandler);
```

Comment:
it looks like you are calling `setMethodCallHandler` multiple times. I think this will cause the previous handler for that channel to be overwritten.

Reply:
You're right, though each instance of the platform has its own copy of the MethodChannel (I HOPE)

Reply:
@hterkelsen you were right, even across different instances of a MethodChannel, only the LAST call to `setMethodCallHandler` seems to be called.

@mklim said: "There's only one handler per channel. A way to thin the Handler code is to actually move the implementation of the methods into other classes, and write the Handler itself just to plumb from method call and response to the business logic and back."